dashboard/app/controllers/api/v1/projects/personal_projects_controller.rb

```diff
@@ -6,4 +8,19 @@ def index
   rescue ArgumentError => e
     render json: {error: e.message}, status: :bad_request
   end
+
+  # GET /api/v1/projects/personal/check_name?new_name=NewName
+  def check_name
+    begin
+      name_failure = ShareFiltering.find_name_failure(params[:new_name], locale)
+    rescue OpenURI::HTTPError, IO::EAGAINWaitReadable => name_checking_error
+      # If WebPurify or Geocoder fail, the name will be allowed, but we log the error
+      slog(
+        tag: 'name_checking_error',
+        error: "#{name_checking_error.class.name}: #{name_checking_error}",
+      )
```

Comment:
I don't know the answer to this, but this looks like a good candidate for logging to Firehose instead.

Reply:
I used `slog` because it's the way we currently log these errors [here](https://github.com/code-dot-org/code-dot-org/blob/d1c5388ae0fa9636e781042076d8a6e76a4e94c2/dashboard/app/controllers/activities_controller.rb#L68). Should I update both places to use firehose?

More info on `slog`:
- https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/config/deployment.rb#L3
- https://github.com/code-dot-org/code-dot-org/blob/staging/lib/cdo.rb#L153
- https://github.com/code-dot-org/code-dot-org/blob/staging/lib/cdo/slog.rb#L14